apps/user-office-backend/src/auth/StfcUserAuthorization.ts

```diff
@@ -226,15 +218,23 @@ export class StfcUserAuthorization extends UserAuthorization {
     return dummyUser;
   }
 
-  async logout(token: string): Promise<void> {
-    this.uowsTokenCache.remove(token);
+  async logout(uosToken: AuthJwtPayload): Promise<void | Rejection> {
+    try {
+      const token = uosToken.externalToken;
+      if (token) {
+        this.uowsTokenCache.remove(token);
 
-    await client.logout(token).catch(() => {
-      logger.logWarn('Failed to log out user', { token });
-      throw new Error(`Failed to logout ${token}`);
-    });
+        await client.logout(token).catch(() => {
+          logger.logWarn('Failed to log out user', { token });
 
-    return;
+          return rejection('Failed to log out user', { token });
+        });
+      } else {
+        return rejection('No external token found in JWT', { token });
+      }
+    } catch (error) {
```

Comment:
It looks like that external try/catch block is still there: https://github.com/UserOfficeProject/user-office-core/pull/51/files#diff-3203c957159c45948037f11a5138ff7deaa651464b70bd47c4dc0ef2f1fa6642R483

so I do think it should be removed from here.

Reply:
External try..catch has to be there because it calls 
`verifyToken(token);` which will throw if the token is invalid

I would also leave the inner try..catch because the logout is a public function and can be called from multiple places. The return type is void | Rejection, so from the API point of view, I would expect to receive a back Rejection, instead of the exception if something fails.

From what I can see the flow is kept the same as it was before, which was the point of this small refactor.

What do you think? Shall we keep it as is?